File: eventsource/resources/cors.php
<?php
  $origin = isset($_GET['origin']) ? $_GET['origin'] : $_SERVER['HTTP_ORIGIN'];
  header("Access-Control-Allow-Origin: $origin");

  $credentials = isset($_GET['credentials']) ? $_GET['credentials'] : "true";
  if ($credentials != "none")
    header("Access-Control-Allow-Credentials: $credentials");

  switch($_GET['run']) {
    case "status-reconnect.php":
    case "message.php":
    case "redirect.php":
    case "cache-control.php":
      include($_GET['run']);
      break;
  }
?>

File: eventsource/resources/cors.py
import os

def run_other(request, response, path):
    #This is a terrible hack
    environ = {"__file__": path}
    execfile(path, environ, environ)
    rv = environ["main"](request, response)
    return rv

def main(request, response):

    print request.request_line
    print request._raw_headers

    origin = request.GET.first("origin", request.headers["origin"])
    credentials = request.GET.first("credentials", "true")

    response.headers.update([("Access-Control-Allow-Origin", origin),
                             ("Access-Control-Allow-Credentials", credentials)])

    handler = request.GET.first('run')
    if handler in ["status-reconnect",
                   "message",
                   "redirect",
                   "cache-control"]:
        if handler == "cache-control":
            return open("cache-control.event_source").read()
        elif handler == "redirect":
            return run_other(request, response, os.path.join(request.doc_root, "common", "redirect.py"))
        else:
            return run_other(request, response, os.path.join(os.path.split(__file__)[0], handler + ".py"))
    else:
        return
